### PR TITLE
fix(TDI-40061): avoid to reset the url factory

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the problem this Pull Request is trying to solve?**
 
**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDI-40061
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/107/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/107/)

File: daikon/src/test/java/org/talend/daikon/runtime/RuntimeUtilTest.java [hunk 1]
```diff
@@ -19,5 +19,11 @@ public void testMvnUrlParseUrl() throws MalformedURLException, IOException {
         URL badurl = new URL(url, "foo");
         assertEquals("org/art/1.2", badurl.getPath());
     }
+    
+    @Test
+    public void testRegisterMavenUrlFactory() throws Exception {
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply daikon/src/test/java/org/talend/daikon/runtime/RuntimeUtilTest.java]: thanks, codacy, though you are a robot, but i like you, happy new year

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/108/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/108/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/112/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/112/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/113/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/113/)

**Pull request merged**